Sources/Datadog/RUM/RUMMonitor/Scopes/RUMViewScope.swift

```diff
@@ -76,9 +76,11 @@ internal class RUMViewScope: RUMScope, RUMContextProvider {
 
     var context: RUMContext {
         var context = parent.context
-        context.activeViewID = viewUUID
-        context.activeViewURI = viewURI
-        context.activeUserActionID = userActionScope?.actionUUID
+        if isActiveView {
```

Comment:
@ncreated @buranmert this change made sense to me, and tests are fine with it.
Could you think of bad consequences such a small change of behaviour could cause though? Maybe I'm missing some context about why `isActiveView` was not accounted for before, and maybe my change doesn't have any effect at all.

Reply:
i guess it was working as `viewScopes.last` is always the _active view_
whatever the other `viewScope`s set as `activeViewID`, it was always corrected by `viewScopes.last`
so it would probably break if `viewScopes` were let's say a `Set` or anything unordered

your change totally makes sense to me 👍

Reply:
I commented on this in https://github.com/DataDog/dd-sdk-ios/pull/376#discussion_r560060267. In short - IMO it doesn't break anything nor doesn't add anything new. In such case, I'd keep the code simple and revert to the previous implementation. We can consider it if there's a reason or the requirements change.

Reply:
if `context.activeView*` is not neccessarily of current `activeView`'s, maybe it shouldn't be called `context.activeView*`, wdyt?

Reply:
This is just the naming thing to keep the lingo consistent with `dd-sdk-android`.

Indeed, `context.viewID` may work better in scopes, but it won't do well for usages in `Logger` or `Tracer`, where we retrieve `contextProvider.context.activeViewID` (which sounds IMO cleaner than `.viewID`). I'd keep it as it is for now, as `RUMM-786` is too small for architecture-wise refactoring.

Reply:
Yeah no worries, I wouldn't take on a refactoring inside such a small scope task :)
Also if this doesn't add any value right now, I'm ok reverting it. Still that would be because we can assume some pattern of calls.
That also means this might be the kind of details that could come back to bite us in a while once we start querying scopes differently for some unrelated reason, and I'm on fence about what to do. The very least would be a comment/TODO, what do you think?

Reply:
I think we can leave a comment in `RUMUserActionScope`, where it retrieves the `context` from `parent` (so the view scope). Right now, it never reads `parent.context` of an inactive view scope, but this may indeed change due to new requirements or so 👍.